Source/WebCore/html/HTMLAttachmentElement.h

```diff
@@ -101,12 +109,19 @@ class HTMLAttachmentElement final : public HTMLElement {
     bool childShouldCreateRenderer(const Node&) const final;
 #endif
 
+    enum class Implementation: uint8_t { Legacy, Modern, LegacyUnderModern };
```

Comment:
Still not a fan of the `LegacyUnderModern` name.

You mentioned:

> It could be more than an icon: Thumbnail, placeholder...

Unless I'm missing something, these are just icons right? A thumbnail icon, or a placeholder icon?

Reply:
Oh, for me, a "thumbnail" is an image that is a small representation of the **contents** of a document (like a minimized screenshot, or an extract), while an "icon" is also an image but an abstract representation of a **type** of documents (like the same PDF logo for all pdf files). Hence my confusion and resistance when you suggested "IconOnly".
Wikipedia seems to agree with me, however I've found another definition of thumbnail saying "an image of a graphic or document that is reduced in size so that it serves as an icon of a larger image", so I'm not so sure anymore.
In any case it shows that it could be confusing to *some* people! 😉

So how about we call it "ImageOnly"?
It should be less prone to confusion, and will better match what we're doing in the old rendering code (tests that skip rendering anything else than images/icons/thumbnails/placeholders).

Reply:
Sounds good to me!